tooling/site-launch/create-searchsg-client.ts

```diff
@@ -0,0 +1,225 @@
+
+  // TODO: add validation via zod
+  const primary = (siteConfig as any).colors?.brand?.canvas?.inverse
+
+  const { displayedName, dataDomain } = await askForDomainAndName({
+    domain,
+    name,
+  })
+
+  const applicationId = await createSearchSgClient({
+    dataDomain,
+    displayedName,
+    domain,
+    primary,
+  })
+
+  siteConfig.site.url = `https://${domain}`
+  siteConfig.site.search = {
+    type: "searchSG",
+    clientId: applicationId,
+  }
+
+  await updateSiteConfig(repo, siteConfig, sha)
+  await addSearchJson(repo)
+}
+
+interface SearchSgConfig {
+  dataDomain: string
+  domain: string
+  displayedName: string
+  primary: string
+}
+const createSearchSgClient = async ({
+  dataDomain,
+  domain,
+  displayedName,
+  primary,
+}: SearchSgConfig) => {
+  const authToken = process.env.SEARCHSG_API_KEY
+
+  const {
+    data: { accessToken, tokenType },
+  } = await axios.post(
+    SearchSgApi.auth,
+    {},
+    {
+      headers: {
+        Authorization: `Basic ${authToken}`,
+        UserAgent: ISOMER_UA,
+      },
+    },
+  )
+
+  const data = generateRequestData({
+    domain: dataDomain,
+    name: displayedName,
+    siteDomain: domain,
+    primary,
+  })
+
+  const {
+    data: {
+      data: {
+        application: { applicationId },
+      },
+    },
+  } = await axios.post(SearchSgApi.create, data, {
+    headers: {
+      "Content-Type": "application/json",
+      "User-Agent": ISOMER_UA,
+      Authorization: `${tokenType} ${accessToken}`,
+    },
+  })
+
+  console.log("Created search sg application with id:", applicationId)
+
+  return applicationId
+}
+
+export const createSearchSgClientForStudio = async ({
+  domain,
+  name,
+}: CreateSearchSgClientParams) => {
+  // TODO: validate this to be #[0-9A-F]{6}
+  const primary = await input({
+    message:
+      "Enter the hex value (eg: #FFFFFF) of the site's `colors.brand.canvas.inverse`:",
+  })
+
+  const { displayedName, dataDomain } = await askForDomainAndName({
+    domain,
+    name,
+  })
+
+  const applicationId = await createSearchSgClient({
+    dataDomain,
+    displayedName,
+    domain,
+    primary,
+  })
+
+  const searchConfigJson = {
+    search: {
+      type: "searchSG",
+      clientId: applicationId,
+    },
+  }
+
+  const urlJson = {
+    url: `https://${domain}`,
+  }
+
+  const searchJsonRelativePath = `${domain}.search.json`
+  writeFileSync(`./${searchJsonRelativePath}`, JSON.stringify(searchConfigJson))
+
+  const urlRelativePath = `${domain}.url.json`
+  writeFileSync(`./${urlRelativePath}`, JSON.stringify(urlJson))
+
+  // TODO: Write to db
+  await confirm({
+    message: `Have you added the contents of ${searchJsonRelativePath} to Site.config?`,
+  })
+  await confirm({
+    message: `Have you added the contents of ${urlRelativePath} to Site.config?`,
+  })
+  await confirm({
+    message: `Have you added the contents of \`search.json\` to the site via Admin mode on Studio?`,
+  })
```

Comment:
issue: will be replaced by writing to DB directly in the future but for now actually the search page cannot be edited via Admin mode cos it is not registered in our schema, so cannot save